javatests/arcs/core/host/HandleAdapterTest.kt

```diff
@@ -109,13 +110,79 @@ class HandleAdapterTest {
         assertThat(writeOnlyHandle).isNotInstanceOf(ReadWriteCollectionHandle::class.java)
     }
 
+    @Test
+    fun singletonHandleAdapter_onUpdateTest() = runBlockingTest {
+        val handle = manager.createSingletonHandle(
+            HandleMode.ReadWrite,
+            READ_WRITE_HANDLE,
+            Person,
+            STORAGE_KEY,
+            Person.SCHEMA
+        ) as ReadWriteSingletonHandle<Person>
+
+        var x = 0
+        handle.onUpdate { x = 1 }
+        handle.store(Person())
+        assertThat(x).isEqualTo(1)
+    }
+
+    @Test
+    fun collectionHandleAdapter_onUpdateTest() = runBlockingTest {
+        val handle = manager.createCollectionHandle(
+            HandleMode.ReadWrite,
+            READ_WRITE_HANDLE,
+            Person,
+            STORAGE_KEY,
+            Person.SCHEMA
+        ) as ReadWriteCollectionHandle<Person>
+
+        var x = 0
+        handle.onUpdate { x = 1 }
+        handle.store(Person())
+        assertThat(x).isEqualTo(1)
+    }
+
+    @Test
+    fun handleAdapter_combineUpdatesTest() = runBlockingTest {
+        val collection = manager.createSingletonHandle(
+            HandleMode.ReadWrite,
+            READ_WRITE_HANDLE,
+            Person,
+            STORAGE_KEY,
+            Person.SCHEMA
+        ) as ReadWriteSingletonHandle<Person>
+
+        val singleton = manager.createCollectionHandle(
+            HandleMode.ReadWrite,
+            READ_WRITE_HANDLE,
+            Person,
+            KEY_TWO,
+            Person.SCHEMA
+        ) as ReadWriteCollectionHandle<Person>
+
+        var x = 0
+        combineUpdates(collection, singleton) { _, _ ->
+            x = x + 1
+        }
+        singleton.store(Person())
+        assertThat(x).isEqualTo(1)
+        collection.store(Person())
+        assertThat(x).isEqualTo(2)
+    }
+
     private companion object {
         private const val READ_ONLY_HANDLE = "readOnlyHandle"
         private const val WRITE_ONLY_HANDLE = "writeOnlyHandle"
+        private const val READ_WRITE_HANDLE = "readWriteHandle"
 
         private val STORAGE_KEY = ReferenceModeStorageKey(
             backingKey = RamDiskStorageKey("backing"),
             storageKey = RamDiskStorageKey("entity")
         )
+
+        private val KEY_TWO = ReferenceModeStorageKey(
```

Comment:
rename these to STORAGE_KEY1 and STORAGE_KEY2?

Reply:
Fixed.